test/messages.ts

```diff
@@ -96,80 +96,80 @@ describe('message content', function() {
         assert.equal(message.body.content[2][0], 'pi');
         assert.equal(message.body.content[2][1], 3.14);
     }));
-    it('sends and receives subject', transfer_test({subject:'my-subject'}, function(message) {
+    it('sends and receives subject', transfer_test({subject:'my-subject'}, function(message: rhea.AmqpMessage) {
         assert.equal(message.subject, 'my-subject');
     }));
-    it('sends and receives message-id as string', transfer_test({message_id:'my-id'}, function(message) {
+    it('sends and receives message-id as string', transfer_test({message_id:'my-id'}, function(message: rhea.AmqpMessage) {
         assert.equal(message.message_id, 'my-id');
     }));
-    it('sends and receives message-id as long', transfer_test({message_id:12345}, function(message) {
+    it('sends and receives message-id as long', transfer_test({message_id:12345}, function(message: rhea.AmqpMessage) {
         assert.equal(message.message_id, 12345);
     }));
     var test_uuid = rhea_util.uuid4();
-    it('sends and receives message-id as uuid', transfer_test({message_id:test_uuid}, function(message) {
-        assert.equal(rhea.uuid_to_string(message.message_id), rhea.uuid_to_string(test_uuid));
+    it('sends and receives message-id as uuid', transfer_test({message_id:test_uuid}, function(message: rhea.AmqpMessage) {
+        assert.equal(rhea.uuid_to_string(message.message_id as Buffer), rhea.uuid_to_string(test_uuid));
```

Comment:
I always expected message_id property of an AmqpMessage to be of type string. However, in this case message_id is being passed as an input to [`rhea.uuid_to_string()`](https://github.com/amqp/rhea/blob/master/lib/util.js#L44). This method expects the input parameter to by of type Buffer. Hence to satisfy this test I had to make message_id a union type of `string | Buffer` in the [`AmqpMessage`](https://github.com/amqp/rhea/pull/70/files#diff-44093036943c4295eb3edaaf2d347ee1R308) interface.

Reply:
In the spec its a union of string, number (ulong) or uuid.

Reply:
I see. Then should the type of message_id be defined as `string | number | Buffer` ?